utils/file_utils.py

```diff
@@ -217,6 +218,9 @@ def from_url_or_local(target, destination):
         if not os.path.isfile(target):
             raise RuntimeError(' - E - %s is not a file!' % target)
 
+        # Desired effect is a valid destination.  Is it done already?
+        if os.path.realpath(target) == os.path.realpath(destination):
```

Comment:
Destination can be equal, but user might want to add/overwrite the content of the folder.
Just returning without action doesn't seem like a good idea. Either overwrite or flipout with error.

Reply:
An error is thrown if you try to copy the identical source file on the destination file.  I have it return nothing because that represents "success".

Reply:
Oh wait, we are dealing with a single file here... then it is fair.